source/tank/Tank.hx

```diff
@@ -104,4 +104,11 @@ class Tank extends FlxSprite {
 		cannon.y = y;
 		cannon.angle = aimDegrees;
 	}
+
+	override function kill() {
+		cannon.kill();
+		bullets.kill();
+		alive = false;
+		exists = false;
+	}
```

Comment:
don't necessarily need 

```haxe
alive = false;
exists = false;
```
we should just call
```
super.kill();
```
to do everything our parent does.

Also, `bullets.kill()` might be kinda weird - it means that if you kill an enemy tank their bullets disappear.

Reply:
I didn't even think about super.kill() I was just trying to call kill without it running in an infinite loop so I just added the original functionality at the end. I'll push a fix it with the super version and also remove the bullet part. I was just destroying everything I saw added to the group without thinking about it.